3rdparty/mbedtls/CMakeLists.txt

```diff
@@ -1,146 +1,187 @@
-add_custom_command(TARGET mbedtls-wrap
-    POST_BUILD
-    COMMAND ${CMAKE_AR} r ${BINARY_DIR}/library/libmbedcrypto.a $<TARGET_OBJECTS:oe_mbedtls_hardware_poll>)
-
-# NOTE: These imported libraries must be specified before `oecore`
-# when linking, hence their dependency on it.
-add_library(mbedtls_imported IMPORTED STATIC)
-set_target_properties(mbedtls_imported PROPERTIES
-  IMPORTED_LOCATION ${BINARY_DIR}/library/libmbedtls.a)
-target_link_libraries(mbedtls_imported INTERFACE oecore)
-
-add_library(mbedx509_imported IMPORTED STATIC)
-set_target_properties(mbedx509_imported PROPERTIES
-  IMPORTED_LOCATION ${BINARY_DIR}/library/libmbedx509.a)
-target_link_libraries(mbedx509_imported INTERFACE oecore)
-
-add_library(mbedcrypto_imported IMPORTED STATIC)
-set_target_properties(mbedcrypto_imported PROPERTIES
-  IMPORTED_LOCATION ${BINARY_DIR}/library/libmbedcrypto.a)
-target_link_libraries(mbedcrypto_imported INTERFACE oecore)
-
-# Convenience lib for use in target_link_libraries
-add_library(mbedcrypto INTERFACE)
-
-add_dependencies(mbedcrypto mbedtls-wrap)
-
-target_include_directories(mbedcrypto INTERFACE
-  $<BUILD_INTERFACE:${SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty>)
-
-target_link_libraries(mbedcrypto INTERFACE
-  $<BUILD_INTERFACE:mbedtls_imported>
-  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_LIBDIR}/openenclave/enclave/libmbedtls.a>
-  $<BUILD_INTERFACE:mbedx509_imported>
-  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_LIBDIR}/openenclave/enclave/libmbedx509.a>
-  $<BUILD_INTERFACE:mbedcrypto_imported>
-  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_LIBDIR}/openenclave/enclave/libmbedcrypto.a>
-  # NOTE: This dependency is repeated because during installation the
-  # imported libraries are not part of the CMake dependency graph, so
-  # this transitive dependency would be left out.
-  oecore
-  oe_includes)
+if (CMAKE_COMPILER_IS_CLANG)
+  list(APPEND MBEDTLS_COMPILE_OPTS
+    -Wpointer-arith
+    -Wimplicit-fallthrough
+    -Wmissing-declarations
+    -Wmissing-prototypes
+    -Wdocumentation
+    -Wno-documentation-deprecated-sync
+    -Wunreachable-code)
+endif()
 
-install(TARGETS mbedcrypto EXPORT openenclave-targets)
+if (USE_CLANGW)
+  list(APPEND MBEDTLS_COMPILE_OPTS "-target x86_64-pc-linux")
 
-install(DIRECTORY ${SOURCE_DIR}/include/mbedtls
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty)
+  # Set up the linker.
+  find_program(LLVM_AR "llvm-ar.exe")
+  set(CMAKE_AR ${LLVM_AR})
+endif()
 
-# NOTE: Because these libraries are `IMPORTED`, it is not possible
-# with CMake to install them as targets, so instead they must be
-# installed as files. This is what causes the above problem with the
-# repeated dependency on `oecore`.
-install(FILES ${BINARY_DIR}/library/libmbedtls.a
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/openenclave/enclave)
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
```

Comment:
What ought to be the flags for RelWithDebInfo?
Won't the optimization-level flags be inherited via oelibc dependency?

Reply:
your right. no need to have this part.